### ISSUE TITLE
[RFC]: Add XPress training pipeline support

### ISSUE THREAD

**Issue by ZKBig** [vllm-project/speculators#1051]
### Motivation.

A block-diffusion drafter such as DFlash proposes a whole block of draft tokens in one forward pass, which makes drafting cheap. The cost is accuracy: the block is decoded from per-position marginals, so token *k* is drawn without seeing what the drafter chose at *k-1*. Nothing ties the positions together, while the target model verifies conditionally, so blocks that are individually plausible but jointly improbable get rejected early. That caps the achievable acceptance length.

We would like to contribute XPress, which restores the missing causality without a token-by-token loop, as a first-class speculator alongside DFlash and DSpark rather than as a fork. On Qwen3-8B it raises acceptance length over the Markov head on every benchmark we measured, at both temperatures, and over the plain DFlash drafter it is worth about +30% acceptance and ~1.3x end-to-end throughput on average. Full tables are in the third field.

Paper: xPress: Parallel Refinement for Diffusion Drafters in Speculative Decoding (https://arxiv.org/abs/2608.02438).

### Proposed Change.

A new speculator, `xpress`, registered next to the existing ones.

**The refiner head.** Per position it reads the drafter's hidden state, a block-global summary and the previous token's embedding, fuses them into a low-rank space (r = 256), mixes across positions with a lower-triangular (causal) linear mixer, passes through an r-space SwiGLU MLP, and reads back out as a bias on the target LM head's logits. Everything except the two vocabulary projections lives in the bottleneck (~80M params, 96% of which are those two matrices). The whole refiner is:

```python
lat = self.token_embed(prev_token_ids)                    # [N, B, r]
x   = self.in_proj(torch.cat([hcache, lat], dim=-1))      # 3r -> r
l_eff = self.mix_l * self.causal_tril                     # lower-triangular, per channel
x = x + torch.einsum("ckj,njc->nkc", l_eff, x)            # causal mix across the block
x = x + self.mlp_down(F.silu(self.mlp_gate(x)) * self.mlp_up(x))
return self.readout(x)                                    # r -> V, a bias on the base logits
```

`hcache` is the pass-invariant `[down_h(h) ; down_g(g)]`, computed once per block and reused. `mix_l` is initialised to zeros, so at init the mixer is the identity and the head starts as a pure previous-token bias.

**Parallel refinement.** Because the mixer is causal, the block is resolved by Jacobi iteration rather than a left-to-right loop: the same call runs K times, each pass re-feeding the previous pass's argmax, so all positions update at once and the serial depth is K (a small constant) instead of the block size. K is a runtime knob, and the accuracy/latency trade-off it exposes is the main reason this is not simply a wider Markov head.

**Co-training.** The refiner is trained together with the drafter. Beyond the teacher-forced term, free-running Jacobi consistency rounds feed the refiner its own argmax and an anchor term holds the co-trained drafter's own logits to the target, so the Jacobi seed does not decay over a long run.

**Shared code, all opt-in.** The DFlash backbone can optionally return its anchor-validity mask behind a keyword argument, leaving the five-element return and every existing caller, subclass and test untouched. Two default-off DFlash config behaviours (an anchor cap and a prefix validity mask) and three general training options that are not XPress-specific: a separate validation corpus, one-conversation-per-batch batching, and a learning-rate scale for a warm-started backbone relative to a cold-started head. With defaults unchanged, no existing model's behaviour changes.

**Suggested order**, most general to most XPress-specific, rather than one large PR:

1. The three general training options above.
2. The opt-in DFlash backbone changes.
3. The XPress speculator itself, with unit tests in the style of the existing DSpark ones.

### Any Other Things.

**Results.** Qwen3-8B with a DFlash block-16 drafter, single-sequence decoding, at most 2048 generated tokens, one H200. `Sp.` is speedup over autoregressive decoding, tau is acceptance length. Each entry uses the throughput-optimal K for that setting. 

Temperature = 0:

| | GSM8K | MATH-500 | AIME25 | HumanEval | MBPP | LCB | MT-Bench | Avg. |
|---|---|---|---|---|---|---|---|---|
| DFlash tau | 6.48 | 7.71 | 7.10 | 6.44 | 5.75 | 7.11 | 3.18 | 6.25 |
| Markov head tau | 9.67 | 9.24 | 7.95 | 7.76 | 6.90 | 7.85 | 4.13 | 7.64 |
| **XPress tau** | **10.11** | **9.62** | **8.35** | **8.15** | **7.11** | **8.40** | **4.38** | **8.02** |
| XPress Sp. | 8.2x | 7.5x | 5.8x | 6.6x | 5.9x | 6.1x | 3.5x | 6.2x |
| Sp. over DFlash | 1.70x | 1.10x | 1.12x | 1.32x | 1.25x | 1.17x | 1.36x | 1.29x |

Temperature = 1 (lossless sampling, tau averaged over 5 seeds):

| | GSM8K | MATH-500 | AIME25 | HumanEval | MBPP | LCB | MT-Bench | Avg. |
|---|---|---|---|---|---|---|---|---|
| DFlash tau | 5.83 | 5.71 | 4.35 | 5.40 | 4.83 | 5.58 | 2.90 | 4.94 |
| Markov head tau | 8.76 | 7.60 | 6.22 | 6.99 | 6.39 | 6.43 | 3.97 | 6.62 |
| **XPress tau** | **9.20** | **8.12** | **6.68** | **7.36** | **6.68** | **6.65** | **4.13** | **6.97** |
| Sp. over DFlash | 1.60x | 1.58x | 1.48x | 1.46x | 1.51x | 1.23x | 1.39x | 1.46x |

**One design question we would rather settle first.** The refiner needs the backbone's anchor-validity mask, which `DFlashDraftModel._backbone_forward` computes but does not return. We currently expose it behind a keyword so nothing existing changes:

```python
def _backbone_forward(self, ..., return_anchor_valid: bool = False, **kwargs):
    ...
    if return_anchor_valid:
        return hidden, logits, targets, aligned_loss_mask, anchored_block_indices, anchor_valid
    return hidden, logits, targets, aligned_loss_mask, anchored_block_indices
```

The alternative is to have the backbone return a structured object instead of a bare tuple. That is cleaner, but it touches every caller and subclass, so we did not assume it. It does not change much of the diff, but it does change shared code DSpark depends on.
